Libs/MRML/Logic/vtkMRMLColorLogic.h

```diff
@@ -39,6 +39,57 @@ class vtkMRMLColorTableNode;
 class VTK_MRML_LOGIC_EXPORT vtkMRMLColorLogic : public vtkMRMLAbstractLogic
 {
 public:
+  typedef struct
+    {
+      std::string CodeValue;
+      std::string CodingSchemeDesignator;
+      std::string CodeMeaning;
+
+      void StandardTerm() {
+        CodeValue = "";
+        CodingSchemeDesignator = "";
+        CodeMeaning = "";
+      }
+      void StandardTerm(std::string value, std::string designator, std::string meaning) {
+        CodeValue = value;
+        CodingSchemeDesignator = designator;
+        CodeMeaning = meaning;
+      }
+
+      void PrintSelf(ostream &os) {
+        os << "\tCode value: " << CodeValue
+           << "\n\tCode scheme designator: " << CodingSchemeDesignator
+           << "\n\tCode meaning: " << CodeMeaning
+           << std::endl;
+      }
+    } StandardTerm;
+
+  struct ColorLabelCategorization
+    {
+    unsigned LabelValue;
+    StandardTerm SegmentedPropertyCategory;
+    StandardTerm SegmentedPropertyType;
+    StandardTerm SegmentedPropertyTypeModifier;
+    StandardTerm AnatomicRegion;
+    StandardTerm AnatomicRegionModifier;
+
```

Comment:
What about adding a method:

```
std::string GetStandardTermProperty(int standardTermId, int propertyId)
```

where 

`standardTermId` 

would be any of 

```
enum StandardTermIdentifiers
{
  SegmentedPropertyCategory = 0,
  SegmentedPropertyType,
  SegmentedPropertyTypeModifier,
  AnatomicRegion,
  AnatomicRegionModifier
}
```

and 

`propertyId`

would be any of 

```
enum StandardTermIdentifiers
{
  CodeValue = 0,
  CodingSchemeDesignator,
  CodeMeaning
}
```


This approach would then allow you to remove these methods:

```
GetSegmentedPropertyCategoryCodeValue
GetSegmentedPropertyCategoryCodeMeaning
GetSegmentedPropertyCategory

GetSegmentedPropertyTypeCodeValue
GetSegmentedPropertyTypeCodeMeaning
GetSegmentedPropertyTypeCodingSchemeDesignator

GetSegmentedPropertyTypeModifierCodeValue
GetSegmentedPropertyTypeModifierCodeMeaning
GetSegmentedPropertyTypeModifierCodingSchemeDesignator

[...]
```

and simply have a method like this one:

```
std::string vtkMRMLColorLogic::GetStandardTermPropertyByLabel(int standardTermIdentifier, int standardTermProperty, int label, const char *lutName)
{

  ColorLabelCategorization labelCat = 
    GetCategorizationByLabel(int label, const char *lutName = "GenericAnatomyColors");

  return labelCat.GetStandardTermProperty(standardTermIdentifier, standardTermProperty);
}
```

And since enum are wrapped in python, that would also work nicely from python.

Testing would also be easier since you can loop through idenitifier and property.



Similarly, you could add:

```
std::string GetStandardTerm(int standardTermIdentifier,  int label, const char *lutName)
std::string GetStandardTermModifier(int standardTermIdentifier,  int label, const char *lutName)
```

to respectively remove method like:

```
GetSegmentedPropertyType()
GetAnatomicRegion()
[...]
```
 and

```
GetSegmentedPropertyTypeModifier()
GetAnatomicRegionModifier()
[...]
```

Overall it means that the concept would be accessed using the enum , stored using struct (these could even be moved into the cpp file as an implementation details into a `vtkInternal` class.

Your API would then be fully wrapped from python (thanks to enum)